src/workerd/api/queue.c++

```diff
@@ -453,7 +453,9 @@ jsg::Ref<QueueEvent> startQueueEvent(
                           jsg::alloc<QueueController>(event.addRef()),
                           jsg::JsValue(h.env.getHandle(js)).addRef(js),
                           h.getCtx());
-      event->waitUntil(kj::mv(promise));
+      event->waitUntil(promise.then(
+        [event=event.addRef()]() mutable { event->setCompletionStatus(QueueEvent::CompletedSuccessfully{}); },
```

Comment:
Sorry for not mentioning it yesterday, but we should probably also call `event->setCompletionStatus` down after line 472, since in that case the old-style `addEventListener(...)` synchronous event handler has already run to completion.

I don't think we *really* need to worry about that case since I'm pretty sure the main queue consumer in question isn't using that old script style, but it wouldn't hurt.